### DISCUSSION TITLE
useController doesn't work for object field

### DISCUSSION THREAD

**Discussion by aspirisen** [react-hook-form/react-hook-form#6796] [Q&A]
Hello, I have a form that is splitted into sections, each section has its own root inside form value. In the section header i need to show "Touched" label. 

I am trying to use "useController" over the object to detect if it has touched or dirty fields, but it does not react to changes inside its fields and do not have fieldState.

Here is codesandbox https://codesandbox.io/s/usecontroller-forked-7hbnm?file=/src/App.tsx

**Comment by bluebill1049** [comment]
isn't that working in the csb? maybe i missing the issue here. can you add some more details?

**Reply by aspirisen** [reply]
The issue is the following: we have the following form values shape

```
type UserSectionValues = {
  FirstName: string;
  LastName: string;
};

type FormValues = {
  userSection: UserSectionValues;
};
```

so we have nested object. 

Then i register `FirstName` and `LastName` as usual using the full path `register('userSection.FirstName')` and that's work. 
However, i need to know if some of the values inside `userSection` have been touched. So i assume that if i would use `useController` and the the name will be just the object `userSection` it will react to changes of `userSection.FirstName` and `userSection.LastName` because `userSection` is a parent.

So using this approach i should have something like that:
```
  const control = useController({
    name: "userSection",
    control: props.control
  });
```

and then i use `control.fieldState.isTouched` and if some field inside like `userSection.FirstName` and `userSection.LastName` are touched then `control.fieldState.isTouched` will also have `touched` state.
